### ISSUE TITLE
General Feedback

### ISSUE THREAD

**Issue by bonham000** [bonham000/fcc-react-tests-module#202]
- [**Previous General Feedback Issue**](https://github.com/bonham000/fcc-react-tests-module/issues/17)
- **Please use this issue to provide any general feedback, suggestions, or questions on these challenges.**
- **For any specific bugs or concerns, please open a new issue and give it the title of the challenge in question.**

**Comment by nickpapasavvas** [comment]
So one thing I would like to note is that maybe a word of caution should be added that if arrow functions are used in a component as class methods then, at the moment, these can't be tested with mocha out of the box.

**Comment by bonham000** [comment]
@nickolaos77 we mention about the differences from using arrow functions in challenge 26 which describe them as experimental syntax so anyone using them should be aware that they may encounter issues with them. Since these challenges aren't concerned with testing at all I think discussing testing with a specific framework may be a little off-topic.

**Comment by rorm** [comment]
Sorry, I missed this thread when reporting something before. Will post in here from now on.

![screen shot 2017-04-25 at 11 53 34](https://cloud.githubusercontent.com/assets/24595643/25382058/fad4458a-29ad-11e7-86e5-7a28090e570d.png)

Unless I am being completely stupid (which is not unlikely), the code I have written for `React 35` should be correct but doesn't pass the tests.

Also, in this challenge, the provided solution sets the initial state of message to `''` rather than `null` and the `handleEnter()` function definition differs from the base challenge code

Cheers

**Comment by bonham000** [comment]
@rorm the callbacks you pass to the event listener handlers are misspelled: they should be `this.handleKeyPress`.

Also, I fixed the inconsistencies between the seed and solution code and updated the app, thanks for pointing these out!

**Comment by rorm** [comment]
Haha! I knew it. Thank you dyslexia (stupidity?)

**Comment by rorm** [comment]
Another one for you:

![screen shot 2017-04-26 at 10 23 06](https://cloud.githubusercontent.com/assets/24595643/25427425/6e3aa994-2a6a-11e7-9d29-b0a98bf916af.png)

React 47: Code will pass the test even if the list elements do not contain the items from the toDoList array

![screen shot 2017-04-26 at 10 27 09](https://cloud.githubusercontent.com/assets/24595643/25427591/f8779388-2a6a-11e7-88e7-83e33c1d2a4b.png)

**Comment by bonham000** [comment]
@rorm I've made an update to the tests for React 47 which should fix that behavior. Please check it out and let me know if this solves the problem!

**Comment by rorm** [comment]
@bonham000 
I'm afraid there is still the same problem on my end; all the tests are passing. What's odd is that sometimes the tests fail as they should

This is the code that causes the problem:

```JSX
const textAreaStyles = {
	width: 235,
	margin: 5
};

class MyToDoList extends React.Component {
	constructor(props) {
		super(props);
		// change code below this line
		this.state = {
    	userInput: '',
      toDoList: []
    }
		// change code above this line
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleChange = this.handleChange.bind(this);
	}
	handleSubmit() {
		const itemsArray = this.state.userInput.split(',');
		this.setState({
			toDoList: itemsArray
		});
	}
	handleChange(e) {
		this.setState({
			userInput: e.target.value
		});
	}
	render() {
		const items = this.state.toDoList.map(function(item){return <li>item</li>}); // should be {item}
		return (
			<div>
				<textarea
					onChange={this.handleChange}
					value={this.state.userInput}
					style={textAreaStyles}
					placeholder="Separate Items With Commas" /><br />
				<button onClick={this.handleSubmit}>Create List</button>
				<h1>My "To Do" List:</h1>
				<ul>
					{items}
				</ul>
			</div>
		);
	}
};
```

**Comment by bonham000** [comment]
@rorm did you refresh that page that code fails for me?

**Comment by rorm** [comment]
@bonham000 It fails for me now so I may have been imagining things. I think I will go through the whole thing again when I am done so will let you know if it happens again.

**Comment by marigerr2** [comment]
Really nice app!  Thanks very much for creating this.

I'm at Challenge 15 - _Pass an Array as Props_.   I'm wondering if there might be a more intuitive "real-world" example of passing an array as a property.   Currently the solution contains duplicated code that sends the same arrays as a properties but just in different order.   

Excerpt from current solution
```
...
<div>
	<h1>What to wear?!</h1>
	<h3>Today is:</h3>
	{ /* change code below this line */ }
	<Selections
	attire={['casual', 'formal', 'super-casual']}
	shirts={['t-shirt', 'sweatshirt', 'collared shirt']}
	pants={['slacks', 'blue jeans', 'khakis', 'shorts', 'pajama pants']}
	shoes={['dress shoes', 'slippers', 'sneakers']}
	/>
	{ /* change code above this line */ }
	<h3>Tomorrow is:</h3>
	{ /* change code below this line */ }
	<Selections
	attire={['formal', 'super-casual', 'casual']}
	shirts={['collared shirt', 't-shirt', 'sweatshirt']}
	pants={['slacks', 'khakis', 'shorts', 'pajama pants', 'blue jeans']}
	shoes={['dress shoes', 'sneakers', 'slippers']}
	/>
	{ /* change code above this line */ }
</div>
...
```

It was unexpected to me as the solution because it didn't seem to be something that would be used in real-world.  (not uncommon in tutorials of course). 
   
But, I'm wondering if a easier example of passing an array as property would be rendering a To-do list, possibly like below.   I'd be happy to flesh it out more in a pull request for consideration with tests and instructions, if interested.

```
const List= (props) => {  
  const listItems = props.list.map((item) =>  
    <li>{item}</li>  
  );  
  return (  
    <ul>{listItems}</ul>  
  );  
};

class ToDoListCreator extends React.Component {
  constructor(props) {
  	super(props);
  }
  render() {
    return (
      <div>
        <h1>To Do Lists</h1>
        <h3>Today</h3>
        <List
          list={['study', 'exercise', 'buy milk']}
        />
        <h3>Tomorrow </h3>
        <List
          list={['call mom', 'birthday party', 'order concert tickets']}
        />
      </div>
    );
  }
};
```
Renders:

![challenge15todolist](https://user-images.githubusercontent.com/8553417/27739729-e18977fe-5daf-11e7-8d84-d5c349764162.PNG)

**Comment by bonham000** [comment]
@seahik yes, actually, that sounds like a great idea to me. I believe @no-stack-dub-sack wrote this challenge so I'll add him here to take a look at your idea, but, I would say this looks like a useful change and if you want to try and implement it with a PR that would be very appreciated!

**Comment by marigerr2** [comment]
ok great!  I'll hold off for a few days before submitting to see if ok also with @no-stack-dub-sack .

**Comment by tmartres** [comment]
Are there any plans on matching the style to the current standard FCC style ? I would be happy to do that if needed.

**Comment by bonham000** [comment]
@tmartres what do you mean specifically?

**Comment by tmartres** [comment]
@bonham000 I mean the colouring and the page's layout, which are not the same as the other pages of the main FCC website. Unless that's made on purpose ?

**Comment by bonham000** [comment]
@tmartres the intention is that eventually these challenges will be merged into the freeCodeCamp codebase, and will simply run within that platform. Although they are essentially finished with QA now, we're waiting on freeCodeCamp Beta redesign for this to happen, so in the meantime there's really no pressing need to do anything about redesigning the site that's hosting the challenges now.

**Comment by tmartres** [comment]
@bonham000 Ok cool, just a newbie trying to contribute in any way :)

**Comment by rkojima** [comment]
Hi, I don't know if it's just me, but it seems like I'm not able to test code in any of the challenges. Even though I'll click on the solution code button and the code shows up, I can't test code. It also seems like the live preview is not showing anything either.
